test/unit/inbound/process-submit-payment-request.test.js

```diff
@@ -0,0 +1,349 @@
+
+  test('should throw when saveInvoiceLines throws', async () => {
+    saveInvoiceLines.mockRejectedValue(new Error('Database save down issue'))
+
+    const wrapper = async () => {
+      await processSubmitPaymentRequest(paymentRequest)
+    }
+
+    expect(wrapper).rejects.toThrow()
+  })
+
+  test('should throw Error when saveInvoiceLines throws Error', async () => {
+    saveInvoiceLines.mockRejectedValue(new Error('Database save down issue'))
+
+    const wrapper = async () => {
+      await processSubmitPaymentRequest(paymentRequest)
+    }
+
+    expect(wrapper).rejects.toThrow(Error)
+  })
+
+  test('should throw error with "Database save down issue" when saveInvoiceLines throws error with "Database save down issue"', async () => {
+    saveInvoiceLines.mockRejectedValue(new Error('Database save down issue'))
+
+    const wrapper = async () => {
+      await processSubmitPaymentRequest(paymentRequest)
+    }
+
+    expect(wrapper).rejects.toThrow(/^Database save down issue$/)
+  })
+
+  test('should throw when mockTransaction.commit throws', async () => {
+    mockTransaction.commit.mockRejectedValue(new Error('Sequelize transaction commit issue'))
+
+    const wrapper = async () => {
+      await processSubmitPaymentRequest(paymentRequest)
+    }
+
+    expect(wrapper).rejects.toThrow()
+  })
+
+  test('should throw Error when mockTransaction.commit throws Error', async () => {
+    mockTransaction.commit.mockRejectedValue(new Error('Sequelize transaction commit issue'))
+
+    const wrapper = async () => {
+      await processSubmitPaymentRequest(paymentRequest)
+    }
+
+    expect(wrapper).rejects.toThrow(Error)
+  })
+
+  test('should throw error with "Sequelize transaction issue" when mockTransaction.commit throws error with "Sequelize transaction commit issue"', async () => {
+    mockTransaction.commit.mockRejectedValue(new Error('Sequelize transaction commit issue'))
+
+    const wrapper = async () => {
+      await processSubmitPaymentRequest(paymentRequest)
+    }
+
+    expect(wrapper).rejects.toThrow(/^Sequelize transaction commit issue$/)
+  })
+
+  // test('should call mockTransaction.rollback when getPaymentRequestByInvoiceNumber throws', async () => {
+  //   getPaymentRequestByInvoiceNumber.mockReset()
+  //   getPaymentRequestByInvoiceNumber.mockRejectedValue(new Error('Sequelize transaction issue'))
+
+  //   const _ = async () => {
+  //     await processSubmitPaymentRequest(paymentRequest)
+  //   }
+
+  //   expect(mockTransaction.rollback).toHaveBeenCalled()
+  // })
+
+  // test('should call mockTransaction.rollback once when getPaymentRequestByInvoiceNumber throws', async () => {
+  //   getPaymentRequestByInvoiceNumber.mockReset()
+  //   getPaymentRequestByInvoiceNumber.mockRejectedValue(new Error('Database retrieval issue'))
+
+  //   const _ = async () => {
+  //     await processSubmitPaymentRequest(paymentRequest)
+  //   }
+
+  //   expect(mockTransaction.rollback).toHaveBeenCalledTimes(1)
+  // })

```

Comment:
Interestingly, this does not register when internal functions throw 🤷‍♂️

Reply:
Although I should try now I have resolved my `reset` mock issues

Reply:
Can't see anything obviously wrong with them.

There is also a different approach for doing this where you can just change the implementation of `getPaymentRequestByInvoiceNumber` to throw an error rather than the promise rejection.  Could give that a go if this doesn't work.  The test case still covers the same scope but of course is a sacrifice to consistency.

Reply:
I wasn't invoking the function -- ah dear!

`jest` is nice and when you call `expect(wrapper).not.rejects()` or whatever, doing so `jest` calls `wrapper` and deals with exception handling :)